### PR TITLE
Paste/input gist URL for mobile. Fixes #1283

### LINKED ISSUE #1283: Viewer: update mobile ui
- [x] The export button has shifted things around in the header. Need to do some tweaking.

- [x] Also, it would be nice to add an input to paste an entire gist URl into. It's a painful UX to open a report on mobile. We could also support pasting a gist URl on the app, similar to how we support paste of JSON.

### PR DESCRIPTION
**Pull request by ebidel**
R: all

Part of #1283. On mobile, you'll get a url input:

screen shot 2016-12-28 at 10 28 02 am

On mobile/desktop, you'll be able to paste the URL of a gist. Invalid urls will cause a butter bar warning.

### CODE REVIEW

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 1]
```diff
@@ -284,6 +301,44 @@ class LighthouseViewerReport {
   }
 
   /**
+   * Handles changes to the gist url input.
+   */
+  onInputChange(e) {
```
Comment (patrickhulce) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: what does this look like on mobile then after entering the url, keyboard prompts done? should we add our own button?

**Comment by ebidel** [comment]
Yea, it requires some to click outside the input or hit enter on the
keyboard.

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js (hunk 1 above)
Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: Yea, it requires some to click outside the input or hit enter on the
keyboard.

**Comment by ebidel** [comment]
@patrickhulce wdyt?

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 2]
```diff
@@ -284,6 +301,44 @@ class LighthouseViewerReport {
...
+  _loadFromGistURL(url) {
+    try {
+      url = new URL(url);
+
+      if (url.origin !== 'https://gist.github.com') {
+        logger.error('URL was not a gist');
+        return;
+      }
+
+      const id = url.href.substring(url.href.lastIndexOf('/') + 1);
```
Comment (paulirish) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: want to pull it out with a regex instead? seems slightly more robust. 

i think the following would work..:

```js
new URL(location.href).pathname.match(/[a-f0-9]{5,}/)
```

Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: done

**Comment by ebidel** [comment]
PTAL

**Comment by paulirish** [comment]
So the viewer now has a few options:

1. Paste gist URL from clipboard
1. Paste report json from clipboard
1. Drag and drop a report.json file
1. Click, file chooser, select a report.json file

Is that correct? 

If so, I think we could adjust some copy..  Here's a proposal:

* File input placeholder: "Enter full gist url" 
  * > "Paste Gist URL or report JSON"
* Notification on copy: "Report copied to clipboard" 
  * > "Report JSON copied to clipboard"
* Welcome blurb: "Instructions: Click here, drag and drop a json report,
or paste the URL of a saved gist." 
  * > "To view report: Paste report JSON or Gist URL. You can also drag 'n drop a `report.json` file or click to select it."

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js (hunk 1 above)
Comment (paulirish) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: I doubt anyone will hand-type the URL, but those that do will have to do as eric said.

for those on mobile devices that just paste the URL into the field, it'll trigger immediately after paste. (i just tested it out)

**Comment by ebidel** [comment]
Updated the copy. I went wit this:

```
To view report: Paste a report.json or Gist URL.<br>
You can also drag 'n drop the file or click here to select it.
```

I kept the input placeholder as "Enter or paste a Gist URL" because "Paste Gist URL or report JSON here" makes it sound like you can paste a json file into the input.

**Pull request merged**